brain/compartmentReportReader.h

```diff
@@ -0,0 +1,241 @@
+/* Copyright (c) 2017, EPFL/Blue Brain Project
+ *                     Juan Hernando <juan.hernando@epfl.ch>
+ *                     Mohamed-Ghaith Kaabi <mohamed.kaabi@epfl.ch>
+ *
+ * This file is part of Brion <https://github.com/BlueBrain/Brion>
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License version 3.0 as published
+ * by the Free Software Foundation.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+#pragma once
+
+#include "types.h"
+#include <future>
+
+namespace brain
+{
+namespace detail
+{
+class CompartmentReportMapping;
+class CompartmentReportReader;
+class CompartmentReportFrame;
+class CompartmentReportView;
+}
+
+/**
+ * Compartment report meta data.
+ * Meta data for the CompartmentReportReader.
+ */
+struct CompartmentReportMetaData
+{
+    /** The  start time of the report */
+    float startTime = 0;
+
+    /** The  end time of the report */
+    float endTime = 0;
+
+    /** The sampling time interval of the report*/
+    float timeStep = 0;
+
+    /** The  time unit of the report */
+    std::string timeUnit;
+
+    /** The  data unit of the report */
+    std::string dataUnit;
+};
+
+/**
+ * Reader compartment report.
+ *
+ * Following RAII, all readers are ready for use after the creation and will
+ * ensure release of resources upon destruction.
+ *
+ * \ifnot pybind
+ *
+ * \endif
+ */
+class CompartmentReportReader : public boost::noncopyable
+{
+public:
+    /**
+     * Open a report in read mode.
+     * @param uri URI to compartment report
+     * @throw std::runtime_error if compartment report could be opened
+     * for read or it is not valid.
+     * @version 1.0
+     */
+    CompartmentReportReader(const URI& uri);
+    ~CompartmentReportReader();
+
+    /** @return the metadata of the report */
+    const CompartmentReportMetaData& getMetaData() const;
+    /**
+     * Create a view of a subset of neurons. An empty gid
+     * set creates a view containing all the data.
+     * @param gids the neurons of interest
+     * @throw std::runtime_error if invalid GID set.
+     * @version 1.0
+     */
+    CompartmentReportView createView(const GIDSet& gids);
+
+private:
+    std::shared_ptr<detail::CompartmentReportReader> _impl;
+};
+
+/**
+ * Compartment report view.
+ * A view is a facility class to acces to the data of a neurons subset
+ * in the report.
+ */
+class CompartmentReportView
+{
+public:
+    CompartmentReportView(CompartmentReportView&&);
```

Comment:
Not needed here. The copy constructor and the assignment operator are deleted since that the class contains a unique_ptr. So, the move constructor/ass op are automatically selected.